src/main/java/org/opensearch/flowframework/model/WorkflowState.java

```diff
@@ -0,0 +1,466 @@
+            return this;
+        }
+
+        /**
+         * Builder method for adding userOutputs
+         * @param userOutputs userOutputs
+         * @return the Builder object
+         */
+        public Builder userOutputs(Map<String, Object> userOutputs) {
+            this.userOutputs = userOutputs;
+            return this;
+        }
+
+        /**
+         * Builder method for adding resourcesCreated
+         * @param resourcesCreated resourcesCreated
+         * @return the Builder object
+         */
+        public Builder resourcesCreated(Map<String, Object> resourcesCreated) {
+            this.userOutputs = resourcesCreated;
+            return this;
+        }
+
+        /**
+         * Allows building a workflowState
+         * @return WorkflowState workflowState Object containing all needed fields
+         */
+        public WorkflowState build() {
+            WorkflowState workflowState = new WorkflowState();
+            workflowState.workflowId = this.workflowId;
+            workflowState.error = this.error;
+            workflowState.state = this.state;
+            workflowState.provisioningProgress = this.provisioningProgress;
+            workflowState.provisionStartTime = this.provisionStartTime;
+            workflowState.provisionEndTime = this.provisionEndTime;
+            workflowState.user = this.user;
+            workflowState.uiMetadata = this.uiMetadata;
+            workflowState.userOutputs = this.userOutputs;
+            workflowState.resourcesCreated = this.resourcesCreated;
+            return workflowState;
+        }
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        XContentBuilder xContentBuilder = builder.startObject();
+        if (workflowId != null) {
+            xContentBuilder.field(WORKFLOW_ID_FIELD, workflowId);
+        }
+        if (error != null) {
+            xContentBuilder.field(ERROR_FIELD, error);
+        }
+        if (state != null) {
+            xContentBuilder.field(STATE_FIELD, state);
+        }
+        if (provisioningProgress != null) {
+            xContentBuilder.field(PROVISIONING_PROGRESS_FIELD, provisioningProgress);
+        }
+        if (provisionStartTime != null) {
+            xContentBuilder.field(PROVISION_START_TIME_FIELD, provisionStartTime.toEpochMilli());
+        }
+        if (provisionEndTime != null) {
+            xContentBuilder.field(PROVISION_END_TIME_FIELD, provisionEndTime.toEpochMilli());
+        }
+        if (user != null) {
+            xContentBuilder.field(USER_FIELD, user);
+        }
+        if (uiMetadata != null && !uiMetadata.isEmpty()) {
+            xContentBuilder.field(UI_METADATA_FIELD, uiMetadata);
+        }
+        if (userOutputs != null && !userOutputs.isEmpty()) {
+            xContentBuilder.field(USER_OUTPUTS_FIELD, userOutputs);
+        }
+        if (resourcesCreated != null && !resourcesCreated.isEmpty()) {
```

Comment:
```suggestion
        if (!resourcesCreated.isEmpty()) {
```

Reply:
Same as above, at first this will be null as there are no resources created.